packages/flutter/test/material/bottom_sheet_theme_test.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('BottomSheetThemeData copyWith, ==, hashCode basics', () {
+    expect(const BottomSheetThemeData(), const BottomSheetThemeData().copyWith());
+    expect(const BottomSheetThemeData().hashCode, const BottomSheetThemeData().copyWith().hashCode);
+  });
+
+  test('BottomSheetThemeData null fields by default', () {
+    const BottomSheetThemeData snackBarTheme = BottomSheetThemeData();
+    expect(snackBarTheme.backgroundColor, null);
+    expect(snackBarTheme.elevation, null);
+    expect(snackBarTheme.shape, null);
+  });
+
+  testWidgets('Default BottomSheetThemeData debugFillProperties', (WidgetTester tester) async {
+    final DiagnosticPropertiesBuilder builder = DiagnosticPropertiesBuilder();
+    const BottomSheetThemeData().debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+        .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
+        .map((DiagnosticsNode node) => node.toString())
+        .toList();
+
+    expect(description, <String>[]);
+  });
+
+  testWidgets('BottomSheetThemeData implements debugFillProperties', (WidgetTester tester) async {
+    final DiagnosticPropertiesBuilder builder = DiagnosticPropertiesBuilder();
+    BottomSheetThemeData(
+      backgroundColor: const Color(0xFFFFFFFF),
+      elevation: 2.0,
+      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(2.0)),
+    ).debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+        .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
+        .map((DiagnosticsNode node) => node.toString())
+        .toList();
+
+    expect(description, <String>[
+      'backgroundColor: Color(0xffffffff)',
+      'elevation: 2.0',
+      'shape: RoundedRectangleBorder(BorderSide(Color(0xff000000), 0.0, BorderStyle.none), BorderRadius.circular(2.0))',
+    ]);
+  });
+
+  testWidgets('Passing no BottomSheetThemeData returns defaults', (WidgetTester tester) async {
+    final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey<ScaffoldState>();
+
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        key: scaffoldKey,
+        body: const Center(child: Text('body')),
+      ),
+    ));
+
+    showModalBottomSheet<void>(
+      context: scaffoldKey.currentContext,
+      builder: (BuildContext context) {
+        return Container(
+          child: const Text('BottomSheet'),
+        );
+      },
+    );
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
```

Comment:
This can be pumpAndSettle(). Note also: for tests like this shouldn't it be possible to skip `showBottomSheet` and just create a BottomSheet directly?

Here and below.

Reply:
That's true, replaced showBottomSheet calls with simply creating a BottomSheet directly.